### ISSUE TITLE
bug: non-discarded compiler example hangs

### ISSUE THREAD

**Issue by ss2165** [CQCL/guppylang#565]
following test from examples hangs rather than throw expected "linear type not discarded" example

```python

def test_discard_bug(validate):
    from guppylang.prelude import quantum
    bad_module = GuppyModule("test")
    bad_module.load_all(quantum)
    @guppy(bad_module)
    def bad(q: qubit @owned) -> qubit:
        tmp = qubit()
        cx(tmp, q)
        # discard(tmp)  # Compiler hangs if tmp is not explicitly discarded
        return q

```

test included in branch `ss/discard_bug`

**Comment by mark-koch** [comment]
This seems to be a pytest problem rather than a compiler problem (see #569)

**Comment by mark-koch** [comment]
Closed as duplicate of #569
